### PR TITLE
Levelsupermarket

### LINKED ISSUE #79: Implement level 2
The player needs to get to the supermarket, grab some toilet paper and get home again.

### PR DESCRIPTION
**Pull request by maccxs**
Changes 
* cleaned up and commented
* normally also levelgethome should work fine
* changed paths of NPC_AIs a little
* no one should get stuck anymore
* added toilet paper counter
* no blinking paper anymore

### CODE REVIEW

**Comment by holke** [comment]
I like this level.
I take it most of the pathfinding code is copied and i do not need to review it?

One suggestion: When i start the level i am immediately outside of my house and may directly collide with NPCs. This surprised me the first time i played.
I think the experience for the player would be better if we start inside the house and can decide by ourselves  when we leave it.

**Comment by holke** [comment]
I didnt find the counter at first, since du to working on the toilet paper level i was used to looking in the
upper left corner.
And then i looked in the upper right corner and didnt find it. Took me a long time to look down, and i knew that it had to be there.

What i want to say:
Maybe we should not change the position of these counters between levels.
Additionally up is better then down.
However, there is a supermarket in the upper left corner.
Can we put both counters in the upper right corner?

**Comment by maccxs** [comment]
Ja, können wir, auch die Idee mit dem Spieler im Haus ist gut. Und der Ehekot ist nicht von mir. LOL ai Code

**Comment by maccxs** [comment]
Changes
* counter in top right edge
* player in the house, game starts with space key
* desciption of counter in start screen
* merged master

File: YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs [hunk 1]
```diff
@@ -56,28 +55,39 @@ void OnTriggerEnter2D(Collider2D other)
...
+   
                 }
             }
         }
     }
 
+    public void ShowPlayer()
+    {
+        playerRend.sortingLayerName = "Default";
+        playerRend.sortingOrder = 0;
```
Comment (holke) [inline YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs]: We can do this much more unity like with:

```suggestion
        playerRend.enabled = true;
```

Thus, no sorting layer trickery is needed.

File: YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs [hunk 2]
```diff
@@ -56,28 +55,39 @@ void OnTriggerEnter2D(Collider2D other)
...
+    }
+
+    public void UnshowPlayer()
+    {
+        playerRend.sortingLayerName = "Background";
+        playerRend.sortingOrder = 0;
```
Comment (holke) [inline YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs]:
```suggestion
        playerRend.enabled = false;
```
See above.

File: YouVsVirus/Assets/Scenes/levelsupermarket/StartScreenLevelsupermarket.unity [hunk 3]
```diff
@@ -1090,6 +1166,162 @@ MonoBehaviour:
...
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  serializedVersion: 6
+  m_Component:
+  - component: {fileID: 1469294022}
+  - component: {fileID: 1469294024}
+  - component: {fileID: 1469294023}
+  m_Layer: 5
+  m_Name: Text (TMP)
```
Comment (holke) [inline YouVsVirus/Assets/Scenes/levelsupermarket/StartScreenLevelsupermarket.unity]: Can you give this a proper name?
If i am not mistaken this is the text "This is your toilet roll counter. You only need to get it to 1 in this level."
We could call it "Toilet roll counter text"

File: YouVsVirus/Assets/Scenes/levelsupermarket/YouVsVirus_Levelsupermarket.unity [hunk 4]
```diff
@@ -368,6 +368,161 @@ CanvasRenderer:
...
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  serializedVersion: 6
+  m_Component:
+  - component: {fileID: 136817191}
+  - component: {fileID: 136817193}
+  - component: {fileID: 136817192}
+  m_Layer: 5
+  m_Name: 'Text: Congratulations'
```
Comment (holke) [inline YouVsVirus/Assets/Scenes/levelsupermarket/YouVsVirus_Levelsupermarket.unity]: This should also be named differently.
I guess this is the space bar text.

File: YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+<<<<<<< HEAD
+            // in Levelgethome and Levelgethome we want all the smileys to stay the same except for the player
+            if(!(LevelSettings.GetActiveSceneName() == "YouVsVirus_Levelgethome" || 
+               LevelSettings.GetActiveSceneName() == "YouVsVirus_Levelsupermarket") || 
+               this.tag == "Player" )
+=======
+            // in Levelgethome we want all the smileys to stay the same except for the player
+            if(LevelSettings.GetActiveLevelSettings().ShowInfectionStatus == true || this.tag == "Player" )
+>>>>>>> master
```
Comment (holke) [inline YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig]: Oh oh, a failed git merge?

File: YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig [hunk 6]
```diff
@@ -0,0 +1,285 @@
+﻿using System.Collections;
```
Comment (holke) [inline YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig]: Does this file even belong in this commit?
It has the .orig suffix, which i think means that is some kind of backup file?

File: YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig.meta [hunk 7]
```diff
@@ -0,0 +1,7 @@
+fileFormatVersion: 2
```
Comment (holke) [inline YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig.meta]: Same as the HamanBas.cs.orig file. Does this even belong here?

File: YouVsVirus/Assets/Scripts/Components/HumanInstants.cs [hunk 8]
```diff
@@ -48,6 +48,10 @@ void Start()
         {
             // get number of NPCs from main menu
             levelSettings = LevelSettings.GetActiveLevelSettings();
+            // need this in every level
```
Comment (holke) [inline YouVsVirus/Assets/Scripts/Components/HumanInstants.cs]: This comment is misleading.
We do not need in every level that ShowInfectionStatus is true and DayLength is 1.

File: YouVsVirus/Assets/Scripts/Components/InfectionControl.cs.orig [hunk 9]
```diff
@@ -0,0 +1,68 @@
+﻿using System;
```
Comment (holke) [inline YouVsVirus/Assets/Scripts/Components/InfectionControl.cs.orig]: Another orig file. Please delete.

I now know where these come from: If you resolve a merge conflict, git saves the original file with the merge markers (HEAD >>>>> and stuff) as a .orig file.

File: YouVsVirus/Assets/Scripts/Components/InfectionControl.cs.orig.meta [hunk 10]
```diff
@@ -0,0 +1,7 @@
+fileFormatVersion: 2
```
Comment (holke) [inline YouVsVirus/Assets/Scripts/Components/InfectionControl.cs.orig.meta]: orig file

File: YouVsVirus/ProjectSettings/EditorBuildSettings.asset.orig [hunk 11]
```diff
@@ -0,0 +1,66 @@
+%YAML 1.1
```
Comment (holke) [inline YouVsVirus/ProjectSettings/EditorBuildSettings.asset.orig]: orig file

File: YouVsVirus/ProjectSettings/Physics2DSettings.asset.orig [hunk 12]
```diff
@@ -0,0 +1,60 @@
+%YAML 1.1
```
Comment (holke) [inline YouVsVirus/ProjectSettings/Physics2DSettings.asset.orig]: Yet another .orig file.

File: YouVsVirus/ProjectSettings/TagManager.asset.orig [hunk 13]
```diff
@@ -0,0 +1,73 @@
+%YAML 1.1
```
Comment (holke) [inline YouVsVirus/ProjectSettings/TagManager.asset.orig]: orig file

**Review by holke — COMMENTED** [review]
I finally found the time to review the new commits. I feel a little bad for all the comments that i have, but let me say first: This has gotten really good :) I like the level, i like your ideas for the start screen, and it works without errors.
Good work :)

Should be only minor changes though.

In the start screen the "PlayerHousePlayer" Object is within the canvas and the other items (supermarket, toilet roll, pfeil, etc.) are not.
Is there a reason for this?
If not, lets make either all members of the canvas or none.

Also in start screen:
The canvas with the intro text is not centered anymore (as in the other levels).
I suppose this is due to the pictures of the house and supermarket in the left hand corner (which i like very much by the way :) ).
However, these are rather larger and could also be smaller, which may offer a way to put the canvas back in the middle again.
However, if it is not possible to fit everything and have the canvas in the middle, then dont do it, we can live with it.

Can NPCs spawn inside the supermarket or player house? What happens then? Should we use the nonspawnable space here?

**Comment by maccxs** [comment]
> 
> 
> I finally found the time to review the new commits. I feel a little bad for all the comments that i have, but let me say first: This has gotten really good :) I like the level, i like your ideas for the start screen, and it works without errors.
> Good work :)
> 
> Should be only minor changes though.
> 
> In the start screen the "PlayerHousePlayer" Object is within the canvas and the other items (supermarket, toilet roll, pfeil, etc.) are not.
> Is there a reason for this?
> If not, lets make either all members of the canvas or none.
> 
> Also in start screen:
> The canvas with the intro text is not centered anymore (as in the other levels).
> I suppose this is due to the pictures of the house and supermarket in the left hand corner (which i like very much by the way :) ).
> However, these are rather larger and could also be smaller, which may offer a way to put the canvas back in the middle again.
> However, if it is not possible to fit everything and have the canvas in the middle, then dont do it, we can live with it.
> 
> Can NPCs spawn inside the supermarket or player house? What happens then? Should we use the nonspawnable space here?

With af02f81b57835816a54fa13ec30ce358038c8038 included your suggestions. Placed panel at same position as in levelgethome (which is also not center!). But I think it looks better now. Player house has a box collider attached for NPCs. They can spawn there but will be immediately pushed to the side. We could use the nonspawnable space, but in my opinion it also works as is.

File: YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig (hunk 6 above)
Comment (maccxs) [reply YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig:1]: Done with 01ae56cfd4394160555d5557a37c22e2a3be84fd

File: YouVsVirus/Assets/Scenes/levelsupermarket/StartScreenLevelsupermarket.unity (hunk 3 above)
Comment (maccxs) [reply YouVsVirus/Assets/Scenes/levelsupermarket/StartScreenLevelsupermarket.unity]: Renamed with 01ae56cfd4394160555d5557a37c22e2a3be84fd

File: YouVsVirus/Assets/Scenes/levelsupermarket/YouVsVirus_Levelsupermarket.unity (hunk 4 above)
Comment (maccxs) [reply YouVsVirus/Assets/Scenes/levelsupermarket/YouVsVirus_Levelsupermarket.unity]: Renamed with 01ae56cfd4394160555d5557a37c22e2a3be84fd

File: YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig (hunk 5 above)
Comment (maccxs) [reply YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig]: Removed file with d9ea9d61fadffde76c93e19cab7444de1962a7d2

File: YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig (hunk 6 above)
Comment (maccxs) [reply YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig]: Removed file with d9ea9d61fadffde76c93e19cab7444de1962a7d2

File: YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig.meta (hunk 7 above)
Comment (maccxs) [reply YouVsVirus/Assets/Scripts/Components/HumanBase.cs.orig.meta]: Removed file with d9ea9d61fadffde76c93e19cab7444de1962a7d2

File: YouVsVirus/Assets/Scripts/Components/HumanInstants.cs (hunk 8 above)
Comment (maccxs) [reply YouVsVirus/Assets/Scripts/Components/HumanInstants.cs]: Made a better comment c6d91a53a14c2f83b96372c01f182039349181a3

File: YouVsVirus/Assets/Scripts/Components/InfectionControl.cs.orig.meta (hunk 10 above)
Comment (maccxs) [reply YouVsVirus/Assets/Scripts/Components/InfectionControl.cs.orig.meta]: Removed https://github.com/youvsvirus/youvsvirus-unity/commit/d9ea9d61fadffde76c93e19cab7444de1962a7d2

File: YouVsVirus/ProjectSettings/EditorBuildSettings.asset.orig (hunk 11 above)
Comment (maccxs) [reply YouVsVirus/ProjectSettings/EditorBuildSettings.asset.orig]: Removed https://github.com/youvsvirus/youvsvirus-unity/commit/d9ea9d61fadffde76c93e19cab7444de1962a7d2

File: YouVsVirus/ProjectSettings/Physics2DSettings.asset.orig (hunk 12 above)
Comment (maccxs) [reply YouVsVirus/ProjectSettings/Physics2DSettings.asset.orig]: Removed 8fd642e5b5fc3cb9a47b2bc24fba53b7ef49a743

File: YouVsVirus/ProjectSettings/TagManager.asset.orig (hunk 13 above)
Comment (maccxs) [reply YouVsVirus/ProjectSettings/TagManager.asset.orig]: Removed and found even one more :-) 8fd642e5b5fc3cb9a47b2bc24fba53b7ef49a743

File: YouVsVirus/Assets/Scenes/levelsupermarket/StartScreenLevelsupermarket.unity [hunk 14]
```diff
@@ -1178,7 +1185,7 @@ GameObject:
   - component: {fileID: 1469294024}
   - component: {fileID: 1469294023}
   m_Layer: 5
-  m_Name: Text (TMP)
+  m_Name: ToileRollCounterText
```
Comment (holke) [inline YouVsVirus/Assets/Scenes/levelsupermarket/StartScreenLevelsupermarket.unity]:
```suggestion
  m_Name: ToiletRollCounterText
```

**Review by holke — APPROVED** [review]

**Pull request merged**